crates/scheduler/src/call/crypto/hash.rs

```diff
@@ -18,12 +18,18 @@ use digest::{FixedOutput, InvalidLength, KeyInit, Output, Update};
 use generic_array::GenericArray;
 use wasefire_applet_api::crypto::hash::{self as api, Algorithm, Api};
 use wasefire_board_api::{self as board, Api as Board, Support};
-use wasefire_error::Error;
+use wasefire_error::{Code, Error};
 
 use crate::applet::store::{MemoryApi, StoreApi};
 #[cfg(feature = "internal-hash-context")]
 use crate::applet::HashContext;
-use crate::{DispatchSchedulerCall, SchedulerCall, Trap};
+use crate::{DispatchSchedulerCall, Failure, SchedulerCall, Trap};
+
+impl From<InvalidLength> for Failure {
+    fn from(_value: InvalidLength) -> Self {
```

Comment:
nit: to prove we are not hiding/discarding information in this conversion
```suggestion
    fn from(InvalidLength: InvalidLength) -> Self {
```